### ISSUE TITLE
The builtin DHCP client doesn't work

### ISSUE THREAD

**Issue by lparkes** [lparkes/azure-agent#12]
The agent contains a builtin DHCP client that it uses when it can't extract the WireServer endpoint address from the current DHCP lease file. 

I can see the agent shutting down and starting up the system DHCP client and tcpdump shows a DHCP request and response, but the agent never configures itself with the endpoint address.

If I disable management of the system DHCP client then I get error messages in the logs about UDP ports not being available, which makes sense.

I need to check whether the packets that tcpdump sees from the agent's builtin DHCP client, or from the system DHCP client restarting?

**Comment by lparkes** [comment]
I have confirmed that the response is read by the Python code in dhcp.py. 

There is nothing in the log after that verbose output.

```
2024-01-25T22:38:54.953571Z VERBOSE Daemon Daemon Send DHCP request: Setting soc
ket.timeout=10, entering recv
2024-01-25T22:38:54.964877Z VERBOSE Daemon Daemon BytesReceived:0x172
2024-01-25T22:38:54.965708Z VERBOSE Daemon Daemon DHCP response:000000: 02 01 06
 00 20 26 A9 F4  00 00 80 00 00 00 00 00  ................
000010: 0A 00 01 04 A8 3F 81 10  A8 3F 81 10 00 0D 3A D0  ................
000020: 80 9E 00 00 00 00 00 00  00 00 00 00 53 59 44 32  ............SYD2
000030: 37 31 31 31 30 37 31 37  30 35 33 00 00 00 00 00  71110717053.....
000040: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
000050: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
000060: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
000070: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
000080: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
000090: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
0000A0: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
0000B0: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
0000C0: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
0000D0: 00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  ................
0000E0: 00 00 00 00 00 00 00 00  00 00 00 00 63 82 53 63  ............c.Sc
0000F0: 01 04 FF FF FF 00 03 04  0A 00 01 01 06 04 A8 3F  ................
000100: 81 10 0F 33 75 68 6B 79  63 32 6B 75 64 70 6E 65  ...3uhkyc2kudpne
000110: 6C 66 65 66 6D 62 79 62  31 6B 6F 62 69 67 2E 70  lfefmbyb1kobig.p
000120: 78 2E 69 6E 74 65 72 6E  61 6C 2E 63 6C 6F 75 64  x.internal.cloud
000130: 61 70 70 2E 6E 65 74 33  04 FF FF FF FF 35 01 02  app.net3.....5..
000140: 36 04 A8 3F 81 10 3A 04  FF FF FF FF 3B 04 FF FF  6...............
000150: FF FF F5 04 A8 3F 81 10  79 17 00 0A 00 01 01 20  ........y.......
000160: A8 3F 81 10 0A 00 01 01  20 A9 FE A9 FE 0A 00 01  ................
000170: 01 FF                                             ..
```

**Comment by lparkes** [comment]
The problem is that `shellutil.run_command(["/etc/rc.d/dhcpcd", "start"])` doesn't work for some reason with a zombie `sh` process not getting reaped. I'm going to guess that the agent is stuck waiting for that process (or another one) to complete.

**Comment by lparkes** [comment]
Fixed by commit e9e5eb5
